apps/charterafrica/src/lib/ecosystem/utils.js

```diff
@@ -0,0 +1,24 @@
+export function localizeData(data = {}) {
```

Comment:
What is the structure of `data`? I was expecting keys would be `en`/`pt`/`fr`.

Reply:
I have changed this to 

```js
 {
 [key]: {
  en: "",
  pt: "",
  fr: ""
}
}
``` @kilemensi

Reply:
I'm still not 100% sure I get it @koechkevin :
```js
        ...result,
      en: {
        ...result.en,
        [key]: data[key].en || data[key],
      },
```

Why do we need .`..result` and `...result.en` if our data is in `data`?

My understanding is: The output of airtable module is data in the following format: For a field that supports localization, for example `description`, 
```js
  {
    name: "",
    description: {
      en: "",
    }
  }
```
if `localized` is `false`, and:
```js
  {
    name: "",
    description: {
      en: "",
      fr: "",
      pt: ""
    }
  }
```
If `localized` is `true`.

How does this method deal with the above?

Reply:
@kilemensi this method is intended to convert data from 
```js
{
 description: {
  en: "",
  fr: "",
  pt: ""
}
...
}
```

to 

```js
{
 en: {
 description : ""
},
pt: {
 description : ""
},
fr: {
 description : ""
},
}
```

Just because payload doesn't allow create or update with objects on a text field

Reply:
Cool @koechkevin , so there 1 use of this function. But what about those `...result`? Lets take a full example. If my data from Airtable is as 👇🏽, how should I transform it so that Payload accepts it?
```js
{
  name: "Koetch",
  description: {
    en: "Regular Dev",
    fr: "Dev ordinaire",
    pt: "Desenvolvedor regular",
  },
  languages: ["JavaScript", "CSS"],
}
```

Reply:
@kilemensi 
```js
{
  en: {
    name: "Koetch",
    description: "Regular Dev",
    languages: ["JavaScript", "CSS"],
  },
  pt: {
    name: "Koetch",
    description: "Desenvolvedor regular",
    languages: ["JavaScript", "CSS"],
  },
  fr: {
    name: "Koetch",
    description: "Dev ordinaire",
    languages: ["JavaScript", "CSS"],
  },
}
```

Reply:
Then I think you need to pass the locales as well @koechkevin , based on if config.localize is enabled (similar to how we handled it in Airtable). Haven't tested the code but I think something like 👇🏽 should do it
```js
function localize(data, locales) {
    return locales.reduce((localized, locale) => {
        localized[locale] = {};
        Object.keys(data).reduce((localeData, key) => {
            localeData[key] = data[key]?.[locale] ?? data[key];
            return localeData;
        }, localized[locale]);
        return localized;
    }, {});
}
```

PS: I also see `delocalize` function that is not used anywhere, and can't think of why we would need it.

Reply:
@kilemensi this also works fine

Reply:
@kilemensi this also works fine

Reply:
Do you mean `localizeData` also works fine @koechkevin ?

I don't doubt it but:
1. It creates more data than needed (e.g. if `localized` if `false`, we only need `en` but it will create for `fr` and `pt`)
2. It has too much/unnecessary code duplication. The only thing that changes between this nine lines is `en`, `fr`, and `pt`: https://github.com/CodeForAfrica/ui/pull/468/files#diff-a49b86971ff1a23df48a437020db40fb15e8fb658fbdfa5e0f877409e28dc850R14-R25. It can be rewritten in 3 lines or less.

The above suggestion addressed these 2 issues... now, leave me alone! It's my off day.